drv/transceivers-server/src/udp.rs

```diff
@@ -112,11 +94,34 @@ impl ServerImpl {
         tx_data_buf: &mut [u8],
     ) {
         let out_len =
-            match hubpack::deserialize(&rx_data_buf[0..meta.size as usize]) {
-                Ok((msg, data)) => self.handle_message(msg, data, tx_data_buf),
+            // attempt to deserialize the header
+            match hubpack::deserialize::<Header>(rx_data_buf) {
+                Ok((header, data)) => {
+                    // header deserialized successfully, so now attempt to
+                    // deserialize the remaining message
+                    match (meta.size as usize).checked_sub(Header::MAX_SIZE) {
+                        Some(msg_size) => match hubpack::deserialize::<Message>(&data[0..msg_size]) {
+                            // handle the message
+                            Ok((msg, data)) => {
+                                self.handle_message(msg, header.message_id, data, tx_data_buf)
+                            },
+                            // try to tell the host something useful about what happened
+                            Err(e) => {
+                                ringbuf_entry!(Trace::DeserializeError(e));
+                                self.handle_deserialization_error(header, tx_data_buf)
+                            },
+                        },
+                        None => {
+                            ringbuf_entry!(Trace::SizeUnderflowError);
```

Comment:
I think I might call this `TruncatedPacket` or something like that. It's true that the subtraction underflowed, but what this really means is we received a short packet. It also means someone's really not doing UDP correctly, but that's another issue :)